### PR TITLE
Build compatibility matrix tests faster

### PR DESCRIPTION
**Pull request by nohwnd**
## Description
Generate solution for compatibility matrix projects, and build them in parallel.

### CODE REVIEW

File: scripts/build.ps1 [hunk 1]
```diff
@@ -165,101 +165,228 @@ function Invoke-TestAssetsBuild {
         Write-Log ".. .. Build: Source: $TPB_TestAssets_Solution -- add NuGet source"
         Invoke-Exe -IgnoreExitCode 1 $nugetExe -Arguments "sources add -Name ""locally-built-testplatform-packages"" -Source $env:TP_TESTARTIFACTS\packages\ -ConfigFile ""$nugetConfig"""
         Invoke-Exe $dotnetExe -Arguments "build $TPB_TestAssets_Solution --configuration $TPB_Configuration -v:minimal -p:CIBuild=$TPB_CIBuild -p:LocalizedBuild=$TPB_LocalizedBuild -bl:""$env:TP_OUT_DIR\log\$Configuration\TestAssets.binlog"""
+    }
+    finally {
+        Write-Log ".. .. Build: Source: $TPB_TestAssets_Solution -- remove NuGet source"
```
Comment (Evangelink) [inline scripts/build.ps1:170]: I think we have a helper method for "intended log writing". Could you use it instead?

File: scripts/build.ps1 [hunk 2]
```diff
@@ -165,101 +165,228 @@ function Invoke-TestAssetsBuild {
...
-            }
+function Invoke-CompatibilityTestAssetsBuild {
+    # Compatibility matrix build.
+    $dotnetExe = Get-DotNetPath
+    Write-Log "Invoke-CompatibilityTestAssetsBuild: Start test assets build."
+    $timer = Start-Timer
+    $generated = Join-Path (Split-Path -Path $TPB_TestAssets) -ChildPath "GeneratedTestAssets"
+    $generatedSln = Join-Path $generated "CompatibilityTestAssets.sln"
+
+    # Figure out if the versions or the projects to build changed, and if they did not just
```
Comment (Evangelink) [inline scripts/build.ps1:185]:
```suggestion
    # Figure out if the versions or the projects to build changed, and if they did not
```

File: scripts/build.ps1 (hunk 1 above)
Comment (Evangelink) [reply scripts/build.ps1:170]: Hum I cannot find it back. So nevermind this comment.

**Review by Evangelink — APPROVED** [review]

File: scripts/build.ps1 (hunk 1 above)
Comment (nohwnd) [reply scripts/build.ps1:170]: Are you mistaking Write-Log for Write-Host? :)

**Pull request merged**